pulp_container/app/tasks/sync_stages.py

```diff
@@ -82,6 +82,47 @@ def __init__(self, remote, signed_only):
         self.manifest_dcs = []
         self.signature_dcs = []
 
+    def _get_content_data_blocking(self, manifest):
+        saved_artifact = manifest.contentartifact_set.first().artifact
+        raw_data = saved_artifact.file
+        content_data = json.load(raw_data)
+        saved_artifact.file.close()
+        return saved_artifact, content_data, raw_data
```

Comment:
how about we return just just raw_data and later on where it is needed we can json.load it. I am trying to avoid to carry both content_data and raw_data

Reply:
I tried it, but there are some places where I need to use `_get_content_data_blocking` and `_download_and_save_artifact_data`. Specially in the `create_listed_manifest`, where there are statements under the `else` branch that need `content_data`. I will end calling json.load more times than if left where they are now. :sob:

Reply:
it will be written in the code twice - under if and under else branch but in the end of the day *called* it will be once.